Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 opened the pull request:
Fixes #4712:Unaligned radio-button with corresponding text.

I am checking if the type is boolean and if it is, I am wrapping the checkbox and the description in a div.
As the graph interaction,image interaction and fraction input interactions are using the same directive making change in this file should fix for those three interactions.
After fix - 
![image](https://i.imgur.com/OiZHM6x_d.jpg?maxwidth=640&shape=thumb&fidelity=medium)

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

seanlip commented on core/templates/dev/head/components/forms/schema_editors/schema_based_bool_editor_directive.html:
```diff
@@ -1,4 +1,4 @@
-<div class="input-group">
+<div class="input-group" style="float:left;margin-right:4px">
```
This is a generic "boolean editor" component that is used everywhere. You want to float-left based on a specific context. So the float-left styling should be used only in that specific context (perhaps in the wrapper div).

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,12 +29,18 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
     <div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
-      <div class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
-      <div>
+      <div ng-if="customizationArgSpec.schema.type!='bool'" class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
```
- Wrap both this and the next div into a single div that has the ng-if.
- Use !== instead of !=.
- Put spaces on the left and right of the !==.

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,12 +29,18 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
         </schema-based-editor>
         <div style="margin-bottom: 15px;"></div>
       </div>
+      <div ng-if="customizationArgSpec.schema.type=='bool'" class="oppia-interaction-customization-label">
```
Ditto: === instead of ==; put spaces on left and right of ==.

srikar0896 replied:
okay.I will make those changes,thanks for the review.

seanlip reviewed (commented):
Thanks @srikar0896! Some comments.

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 commented:
@seanlip made some changes,please review it.

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,10 +29,20 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
   <form ng-if="stateInteractionIdService.displayed && hasCustomizationArgs" name="form.schemaForm" class="protractor-test-interaction-editor">
     <div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
+      <div ng-if="customizationArgSpec.schema.type !== 'bool'">
       <div class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
```
Should indent this line by 2, right?

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,10 +29,20 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
+        <div>
+          <schema-based-editor local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema">
+          </schema-based-editor>
+          <div style="margin-bottom: 15px;"></div>
```
If this is used in both branches would it be possible to take it out of the ng-ifs and put it below both? The less code that's branched on, the better -- it removes duplication.

Bonus points if you can add a CSS class to the schema-based-editor tag and do it there instead (see below).

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,10 +29,20 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
+        </div>
+      </div>
+      <div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">
+        <div style="float:left;margin-right:4px;">
```
Use a class tailored to this directive. Look at other directives which have style tags at the bottom of the directive, you can follow a similar pattern. See e.g. #4715.

Also see if you can apply some of this styling to the schema-based-editor tag directly. Having all these extra divs is kind of messy. (ng-class may be useful for applying conditional classes.)

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 commented:
@seanlip made some changes [(see)](https://github.com/oppia/oppia/pull/4718/commits/6c26f30c8b658ce8127f06ac806075314b70aa09).
I have added ng-class to the directive itself and added style in the directive.
In the case of Boolean type argument the description have to be after the checkbox.So I am putting two conditions to show argument description in the right place.please review

srikar0896 committed:
Revert "Fix #4712:Unaligned radio-button with corresponding text."
Wrong commit
This reverts commit f80fb289e29e620ec0f0d415423b68c8fcb228b7.

seanlip commented on core/templates/dev/head/components/forms/schema_editors/schema_based_bool_editor_directive.html:
```diff
@@ -4,3 +4,9 @@
            focus-on="<[labelForFocusTarget()]>">
   </span>
 </div>
+<style>
```
Again, this file should not be modified at all. Please keep CSS changes scoped to customize_interaction_modal_directive.html. You can use a specific CSS selector (e.g. `customize-interaction-modal .boolean-checkbox`). The `customize-interaction-modal` ensures that only that directive is affected.

srikar0896 replied:
oh.Okay.got it.Thanks

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,12 +29,13 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
     <div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
-      <div class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
+      <div ng-if="customizationArgSpec.schema.type !== 'bool'" class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
       <div>
```
Do you need this containing div?

srikar0896 replied:
That div only have the description.That div closes in the same line.

seanlip replied:
I'm talking about line 33...

srikar0896 replied:
no,I think removing it does not affect anything.

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -29,12 +29,13 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
+      <div ng-if="customizationArgSpec.schema.type !== 'bool'" class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
       <div>
-        <schema-based-editor local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema">
+        <schema-based-editor ng-class="{'align-checkbox-left': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema">
```
Suggest naming the class `boolean-checkbox` or similar. In general class names should describe the item and not the styling that happens to be applied to it.

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 committed:
Fix #4712:Unaligned radio-button with corresponding text.

srikar0896 commented:
@seanlip followed your suggestions!please review.

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -67,3 +66,9 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
   </button>
 </div>
+<style>
+  .interaction-editor-form .boolean-checkbox {
```
Can you prefix all this with the name of the directive (customize-interaction-modal)? That makes sure the style is scoped to this component only.

@apb7 -- FYI, this might be something that might be worth automating checks for.

srikar0896 replied:
I think we can not name that,because the the html is injected to the body of the modal through template(path to customize_interaction_modal_directive.html). so,there is no <customize -interaction-modal> tag to name in that way.so I added class name for the form for scoping style to that component only.please correct me if I am wrong.

seanlip replied:
Could you please paste here the part of the DOM tree containing this entire component? You can get it by doing "Inspect Element" in your browser and looking at the Elements tab.

srikar0896 replied:
<div uib-modal-window="modal-window" class="modal fade in" role="dialog" index="0" animate="animate" ng-style="{'z-index': 1050 + $$topModalIndex*10, display: 'block'}" tabindex="-1" uib-modal-animation-class="fade" modal-in-class="in" modal-animation="true" style="z-index: 1050; display: block;"><div class="modal-dialog "><div class="modal-content" uib-modal-transclude=""><div class="modal-header" style="border-bottom: 0; padding-bottom: 0;">
  <!---->

  <!----><h3 ng-if="stateInteractionIdService.displayed &amp;&amp; (customizationModalReopened || hasCustomizationArgs)" class="" style="">
    Customize Interaction (Fraction Input)
  </h3><!---->
</div>

<div class="modal-body">
  <!---->

  <!----><form ng-if="stateInteractionIdService.displayed &amp;&amp; hasCustomizationArgs" name="form.schemaForm" class="interaction-editor-form protractor-test-interaction-editor ng-pristine ng-valid" style="">
    <!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema" class="boolean-checkbox"><!---->

<!----><schema-based-bool-editor ng-if="schema().type === 'bool' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" label-for-focus-target="labelForFocusTarget()"><div class="input-group">
  <!----><span ng-if="!expressionMode">
    <input type="checkbox" ng-model="$parent.localValue" ng-disabled="isDisabled()" focus-on="" class="ng-pristine ng-untouched ng-valid ng-empty" aria-invalid="false">
  </span><!---->
</div>
</schema-based-bool-editor><!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!----><div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">Require the learner's answer to be in simplest form</div><!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema" class="boolean-checkbox"><!---->

<!----><schema-based-bool-editor ng-if="schema().type === 'bool' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" label-for-focus-target="labelForFocusTarget()"><div class="input-group">
  <!----><span ng-if="!expressionMode">
    <input type="checkbox" ng-model="$parent.localValue" ng-disabled="isDisabled()" focus-on="" class="ng-pristine ng-untouched ng-valid ng-not-empty" aria-invalid="false">
  </span><!---->
</div>
</schema-based-bool-editor><!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!----><div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">Allow improper fractions in the learner's answer</div><!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema" class="boolean-checkbox"><!---->

<!----><schema-based-bool-editor ng-if="schema().type === 'bool' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" label-for-focus-target="labelForFocusTarget()"><div class="input-group">
  <!----><span ng-if="!expressionMode">
    <input type="checkbox" ng-model="$parent.localValue" ng-disabled="isDisabled()" focus-on="" class="ng-pristine ng-untouched ng-valid ng-not-empty" aria-invalid="false">
  </span><!---->
</div>
</schema-based-bool-editor><!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!----><div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">Allow the answer to contain an integer part</div><!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!----><div ng-if="customizationArgSpec.schema.type !== 'bool'" class="oppia-interaction-customization-label">Custom placeholder text (optional)</div><!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema"><!---->

<!---->

<!---->

<!---->

<!----><schema-based-unicode-editor ng-if="schema().type === 'unicode' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" validators="schema().validators" ui-config="schema().ui_config" label-for-focus-target="labelForFocusTarget()" on-input-blur="onInputBlur" on-input-focus="onInputFocus"><span>
  <!---->

  <!----><span ng-if="!getCodingMode() &amp;&amp; !getRows()" class="" style="">
    <input type="text" ng-disabled="isDisabled()" class="form-control ng-pristine ng-untouched ng-valid ng-empty" ng-model="$parent.localValue" ng-blur="onInputBlur()" ng-focus="onInputFocus()" focus-on="" ng-keypress="onKeypress($event)" placeholder="" apply-validation="validators()" style="display: inline;" aria-invalid="false">
  </span><!---->

  <!-- Note that we do not yet support ng-blur for the code editor. -->
  <!---->
</span>
</schema-based-unicode-editor><!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!---->
  </form><!---->

  <!---->
</div>

<div class="modal-footer">
  <!---->
  <!-- Do not show this button if an interaction is already selected. -->
  <!----><button class="btn btn-default" style="float: left;" ng-click="returnToInteractionSelector()" ng-if="stateInteractionIdService.displayed &amp;&amp; !stateInteractionIdService.savedMemento &amp;&amp; hasCustomizationArgs">Change Interaction</button><!---->

  <!----><button class="btn btn-default" ng-click="cancel()" ng-if="hasCustomizationArgs">Cancel</button><!---->
  <!----><div class="btn-group" ng-if="hasCustomizationArgs" popover-append-to-body="" popover-placement="bottom" popover-trigger="none" ng-attr-uib-popover="<[getSaveInteractionButtonTooltip()]>" uib-popover="" style="">
    <button class="btn btn-success protractor-test-save-interaction" ng-click="save()" ng-disabled="!isSaveInteractionButtonEnabled()">
      Save Interaction
    </button>
  </div><!---->
  <!---->
</div>
<style>
  .interaction-editor-form .boolean-checkbox {
    float: left;
    margin-right: 4px;
  }
</style>
</div></div>
</div>

seanlip replied:
Ah, I see, it's transcluded. OK, thanks! I think this looks good from a code perspective.

Once you get a UI review from @prasanna08 I think you're all set!

srikar0896 replied:
thanks for the review!

seanlip commented:
Adding @prasanna08 for UI review.

prasanna08 reviewed (approved):
Hi @srikar0896 @seanlip, sorry for the late response (I wasn't aware of my assignment since I was receiving all emails during my maintainer week). 
Anyways, I tested UI on my laptop and it looks good. Now checkboxes are aligned with its text.

seanlip commented on core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html:
```diff
@@ -27,14 +27,13 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
-      <div>
-        <schema-based-editor local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema">
+      <div ng-if="customizationArgSpec.schema.type !== 'bool'" class="oppia-interaction-customization-label"><[customizationArgSpec.description]></div>
+        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema">
```
Hi @srikar0896, sorry for missing this earlier, but shouldn't this line and the next one be de-indented by 2?

srikar0896 replied:
@seanlip yes,thanks.Changed and submitted the changes.

srikar0896 committed:
indentation fix.

seanlip reviewed (approved):
LGTM. Thanks!
